File: Games/RockPaperScissors/src/main/java/jokatu/game/games/rockpaperscissors/views/RockPaperScissorsViewResolverFactory.java
package jokatu.game.games.rockpaperscissors.views;

import jokatu.game.games.rockpaperscissors.game.RockPaperScissorsGame;
import jokatu.game.player.StandardPlayer;
import jokatu.game.viewresolver.ViewResolver;
import jokatu.game.viewresolver.ViewResolverFactory;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

/**
 * @author steven
 */
@Component
public class RockPaperScissorsViewResolverFactory extends ViewResolverFactory<StandardPlayer, RockPaperScissorsGame> {
	@NotNull
	@Override
	protected Class<RockPaperScissorsGame> getGameClass() {
		return RockPaperScissorsGame.class;
	}

	@NotNull
	@Override
	protected ViewResolver<StandardPlayer, RockPaperScissorsGame> getResolverFor(@NotNull RockPaperScissorsGame castGame) {
		return new RockPaperScissorsViewResolver(castGame);
	}
}

File: Games/RockPaperScissors/src/main/kotlin/jokatu/game/games/rockpaperscissors/views/RockPaperScissorsViewResolverFactory.kt
package jokatu.game.games.rockpaperscissors.views

import jokatu.game.games.rockpaperscissors.game.RockPaperScissorsGame
import jokatu.game.player.StandardPlayer
import jokatu.game.viewresolver.ViewResolver
import jokatu.game.viewresolver.ViewResolverFactory
import org.springframework.stereotype.Component

/**
 * @author steven
 */
@Component
class RockPaperScissorsViewResolverFactory : ViewResolverFactory<StandardPlayer, RockPaperScissorsGame>(RockPaperScissorsGame::class) {
	override fun getResolverFor(castGame: RockPaperScissorsGame): ViewResolver<StandardPlayer, RockPaperScissorsGame> {
		return RockPaperScissorsViewResolver(castGame)
	}
}
